### PR TITLE
SFD-3376 Integration contract test

### PR DESCRIPTION
**Pull request by ArifNhs**
SFD-3376 Integration contract test

### CODE REVIEW

File: application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/AuthServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import uk.nhs.digital.uec.api.authentication.model.AuthToken;
+import uk.nhs.digital.uec.api.authentication.model.Credential;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class AuthServiceTest {
+
+  @LocalServerPort private int port;
```
Comment (gwaardenburg) [inline application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/AuthServiceTest.java:24]: Can we format these to have the annotation above the variable please.

File: application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/AuthServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  @Autowired public TestRestTemplate restTemplate;
+
```
Comment (gwaardenburg) [inline application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/AuthServiceTest.java:27]: Same as above

File: application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/FuzzySearchSuccessTest.java [hunk 3]
```diff
@@ -34,14 +33,13 @@
   @Autowired private TestRestTemplate restTemplate;
```
Comment (gwaardenburg) [inline application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/FuzzySearchSuccessTest.java:33]: formatting

File: application/app/src/test/resources/application.yaml [hunk 4]
```diff
@@ -44,3 +46,13 @@ cognito:
   userPool.clientId: testUserPoolClientId
   userPool.id: testUserPoolId
   userPool.clientSecret: testUserPoolClientSecret
+
+postcode:
+  mapping:
+    url: https://127.0.0.1:18085/api
```
Comment (gwaardenburg) [inline application/app/src/test/resources/application.yaml:52]: Double check that this is the correct url and port. If we are running the tests in their own container will they be able to see the fuzzy search api/postcode api on local host? Or does it need to be the relative docker network ip?

@jope1 Thoughts?

**Review by gwaardenburg — CHANGES_REQUESTED** [review]

File: application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/AuthServiceTest.java (hunk 1 above)
Comment (ArifNhs) [reply application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/AuthServiceTest.java:24]: vscode does this automatically, during saving the file

File: application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/AuthServiceTest.java (hunk 2 above)
Comment (ArifNhs) [reply application/app/src/test/java/uk/nhs/digital/uec/api/integration/fuzzysearch/AuthServiceTest.java:27]: formatting looks alright and sonarqube doesn't complain.

**Review by gwaardenburg — APPROVED** [review]

**Pull request merged**